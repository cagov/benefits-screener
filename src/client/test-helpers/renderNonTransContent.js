import React from "react";
import { shallow } from "enzyme";
import { I18nextProvider, Trans } from "react-i18next";
import i18n from "./i18nForTests";
import shallowUntilTarget from "./shallowUntilTarget";

function renderNonTransContent(Component, props = {}) {
  const providersWrapper = shallow(
    <I18nextProvider i18n={i18n}>
      <Component {...props} />
    </I18nextProvider>
  );

  const wrapper = shallowUntilTarget(providersWrapper, Component.name);
  // Incorrectly using renderNonTransContent to render a component that
  // contains a Trans component results in a silent failure (the rendered
  // snapshot will contain the Trans component itself instead of the rendered
  // contents). This test makes that silent failure more obvious.
  expect(wrapper.containsMatchingElement(<Trans />)).toBe(false);

  return wrapper;
}

module.exports = renderNonTransContent;
