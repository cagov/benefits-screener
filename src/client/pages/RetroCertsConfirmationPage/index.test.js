import renderNonTransContent from "../../test-helpers/renderNonTransContent";
import Component from "./index";
import AUTH_STRINGS from "../../../data/authStrings";

describe("<RetroCertsConfirmationPage />", () => {
  it("with confirmation number", async () => {
    const wrapper = renderNonTransContent(
      Component,
      "RetroCertsConfirmationPage",
      {
        userData: {
          status: AUTH_STRINGS.statusCode.ok,
          weeksToCertify: [0, 1, 2, 5, 6],
          confirmationNumber: "CONFIRMATION_NUMBER",
        },
        setUserData: () => {},
      }
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("no confirmation number", async () => {
    const wrapper = renderNonTransContent(
      Component,
      "RetroCertsConfirmationPage",
      {
        userData: {
          status: AUTH_STRINGS.statusCode.ok,
          weeksToCertify: [0, 1, 2, 5, 6],
        },
        setUserData: () => {},
      }
    );

    expect(wrapper).toMatchSnapshot();
  });
});
