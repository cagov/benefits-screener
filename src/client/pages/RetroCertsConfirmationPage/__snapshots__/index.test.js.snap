// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RetroCertsConfirmationPage /> no confirmation number 1`] = `
<Redirect
  to="/retroactive-certification/weeks-to-certify"
/>
`;

exports[`<RetroCertsConfirmationPage /> with confirmation number 1`] = `
<div
  id="overflow-wrapper"
>
  <Header />
  <main
    className="pb-5"
    id="certification-page"
  >
    <div
      className="container p-4"
    >
      <h1>
        Retroactive Certification for Unemployment
      </h1>
      <LanguageSelector
        className="mt-3 mb-4"
      />
      <Alert
        className="mt-5"
        closeLabel="Close alert"
        show={true}
        transition={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "appear": false,
              "in": false,
              "mountOnEnter": false,
              "timeout": 300,
              "unmountOnExit": false,
            },
            "displayName": "Fade",
            "render": [Function],
          }
        }
        variant="success"
      >
        <img
          alt="checkmark"
          className="checkmark"
          src="/images/check-circle-fill.svg"
        />
        You have successfully submitted your retroactive certification. No further action is required.
      </Alert>
      <h2
        className="mt-5"
      >
        Confirmation Number
      </h2>
      <p>
        <Trans
          i18nKey="retrocerts-confirmation.p1"
          t={[Function]}
          values={
            Object {
              "confirmationNumber": "_NUMBER",
            }
          }
        />
      </p>
      <p>
        Keep this number for your records.
      </p>
      <p>
        Youâ€™ve been logged out of the system and may close this window.
      </p>
      <Button
        active={false}
        className="text-dark bg-light"
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="outline-secondary"
      >
        Print
      </Button>
      <h2
        className="mt-5"
      >
        Certifications Submitted
      </h2>
      <Button
        active={false}
        className="text-dark bg-light mb-3"
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="outline-secondary"
      >
        Show all
      </Button>
      <ListOfWeeksWithDetail
        showContent={
          Array [
            false,
            false,
            false,
            false,
            false,
            false,
          ]
        }
        toggleContent={[Function]}
        userData={
          Object {
            "confirmationNumber": "CONFIRMATION_NUMBER",
            "programPlan": Array [
              "PUA full time",
            ],
            "status": "ok",
            "weeksToCertify": Array [
              0,
              1,
              2,
              5,
              6,
            ],
          }
        }
      />
      <AccordionItem
        content={
          <AcknowledgementDetail
            className="detail"
          />
        }
        header={
          <strong>
            Acknowledgement
          </strong>
        }
        showContent={false}
        toggleContent={[Function]}
      />
      <h2
        className="mt-5"
      >
        Next Steps
      </h2>
      <p>
        <Trans
          i18nKey="retrocerts-confirmation.p2"
          t={[Function]}
        >
          If you are still unemployed, continue to certify for benefits every two weeks. The fastest way is to use
           
          <a
            href="https://edd.ca.gov/Unemployment/UI_Online.htm"
          >
            UI Online
          </a>
          .
        </Trans>
      </p>
    </div>
  </main>
  <Footer
    backToTopTag="certification-page"
  />
</div>
`;
